Patch: Working Vagrant Bild on PHP7

bobrik717 opened the pull request:
| Q             | A
| ------------- | ---
| Is bugfix?    | yes/no
| New feature?  | yes/no
| Breaks BC?    | yes/no
| Tests pass?   | yes/no
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

bobrik717 commented:
Working Vagrant Bild on PHP7

cebe commented:
can you explain a bit more about what you are doing here? a lot of the changes seem unrelated to php7 and also wrong.

bobrik717 commented:
when deploying the project was a lot of mistakes, especially composer could not Installed all the dependencies because he needed php7. I rewrote sh halyards so vagrant to unfold php7 and added the ability to work under one domain. Now if you deploy the project sufficient to establish a vagrant and virtualbox.

samdark commented on Vagrantfile:
```diff
@@ -2,8 +2,8 @@ require 'yaml'
-  frontend: 'y2aa-frontend.dev',
-  backend:  'y2aa-backend.dev'
+  frontend: 'local.site.com',
+  backend:  'local.site.com/admin'
```
Doesn't look right. Should be 2 separate hosts.

samdark commented on Vagrantfile:
```diff
@@ -18,7 +18,7 @@ options = YAML.load_file config[:local]
 # check github token
 if options['github_token'].nil? || options['github_token'].to_s.length != 40
-  puts "You must place REAL GitHub token into configuration:\n/yii2-app-advanced/vagrant/config/vagrant-local.yml"
+  puts "You must place REAL GitHub token into configuration:\n/yii2-app-advancded/vagrant/config/vagrant-local.yml"
```
Looks like typo introduced.

samdark commented on backend/config/main.php:
```diff
@@ -15,6 +15,7 @@
     'components' => [
         'request' => [
             'csrfParam' => '_csrf-backend',
+            'baseUrl' => '/admin'
```
This isn't needed if hosts are separate.

samdark commented on vagrant/config/vagrant-local.example.yml:
```diff
@@ -10,7 +10,7 @@ timezone: Europe/London
 
 # Virtual machine name
-machine_name: y2aa
+machine_name: site
```
Why?

samdark commented on vagrant/nginx/app.conf:
```diff
@@ -1,77 +1,46 @@
+   location ~* \.(htaccess|htpasswd|svn|git) {
        deny all;
    }
+   location /admin {
```
It should be separate host as it was before.

rugabarbo commented:
PR #236 solves problem of php upgrade. So we can reject and close this PR (#227).

samdark commented:
@bobrik717 sorry for rejecting but it doesn't fit current setup.
